contracts/token/ConfidentialFungibleToken.sol

```diff
@@ -227,11 +225,11 @@ abstract contract ConfidentialFungibleToken is IConfidentialFungibleToken {
     ) public virtual {
         FHE.checkSignatures(requestId, signatures);
 
-        euint64 requestHandle = _requestHandles[requestId];
+        euint64 requestHandle = euint64.wrap(FHE.loadRequestedHandles(requestId)[0]);
         require(FHE.isInitialized(requestHandle), ConfidentialFungibleTokenInvalidGatewayRequest(requestId));
         emit AmountDisclosed(requestHandle, amount);
 
-        _requestHandles[requestId] = euint64.wrap(0);
+        Impl.getDecryptionRequests().requestedHandles[requestId] = new bytes32[](0);
```

Comment:
I think this could be dangerous if the token is extended to add additional functionality (like unwrapping). We have to be sure we understand the implications here.

Reply:
https://github.com/OpenZeppelin/openzeppelin-confidential-contracts/blob/master/contracts/token/extensions/ConfidentialFungibleTokenERC20Wrapper.sol

We also use decryption requests here and if someone calls the finalizeDecyption before finalize unwrap, it would corrupt the unwrap request.